src/utils/mashupDocumentParser.ts

```diff
@@ -150,3 +161,131 @@ export const editSingleQueryMetadata = (metadataArray: Uint8Array, metadata: Met
+            newMetadataXmlSize,
+            newMetadataXml,
+            endBuffer
+        );
+        
+        return newMetadataArray;
+    };
+
+    const updateConnectionOnlyMetadataStr = (metadataString: string, connectionOnlyQueryNames: string[]) => {
+        const parser: DOMParser = new DOMParser();
+        let updatedMetdataString: string = metadataString;         
+        connectionOnlyQueryNames.forEach((queryName: string) => {
+            const metadataDoc: Document = parser.parseFromString(updatedMetdataString, xmlTextResultType);
+            const items: Element = metadataDoc.getElementsByTagName(element.items)[0];
+            const stableEntriesItem: Element = createStableEntriesItem(metadataDoc, queryName);
+            items.appendChild(stableEntriesItem);
+            const sourceItem: Element = createSourceItem(metadataDoc, queryName);
+            items.appendChild(sourceItem);
+            const serializer: XMLSerializer = new XMLSerializer();
+            updatedMetdataString = serializer.serializeToString(metadataDoc);
+        });    
+
+        return updatedMetdataString;
+    };
+
+    const createSourceItem = (metadataDoc: Document, queryName: string) => {
+        const newItemSource: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.item);
+        const newItemLocation: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.itemLocation);
+        const newItemType: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.itemType);
+        newItemType.textContent = "Formula";
+        const newItemPath: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.itemPath);
+        newItemPath.textContent = `Section1/${queryName}/Source`;
+        newItemLocation.appendChild(newItemType);
+        newItemLocation.appendChild(newItemPath);
+        newItemSource.appendChild(newItemLocation);
+        
+        return newItemSource;
+    };
+
+    const createStableEntriesItem = (metadataDoc: Document, queryName: string) => {
+        const newItem: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.item);
+        const newItemLocation: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.itemLocation);
+        const newItemType: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.itemType);
+        newItemType.textContent = "Formula";
+        const newItemPath: Element = metadataDoc.createElementNS(metadataDoc.documentElement.namespaceURI, element.itemPath);
+        newItemPath.textContent = `Section1/${queryName}`;
+        newItemLocation.appendChild(newItemType);
+        newItemLocation.appendChild(newItemPath); 
+        newItem.appendChild(newItemLocation);
+        const stableEntries: Element = createConnectionOnlyEntries(metadataDoc);
```

Comment:
If I'm not missing anything, I think you can just call createSourceItem and then continue with line 230 here?